### ISSUE TITLE
Make owncast.online opengraph tags absolute URLs

### ISSUE THREAD

**Issue by gabek** [owncast/owncast#2230]
### Share your bug report, feature request, or comment.

https://github.com/owncast/owncast.github.io

![image](https://user-images.githubusercontent.com/414923/196819158-7c3edff3-8024-4151-8609-f7702eaab57f.png)

**Comment by Chaitanya-Shahare** [comment]
Can you assign this to me?

**Comment by Chaitanya-Shahare** [comment]
can you assign this to me?

**Comment by gabek** [comment]
Sure, thank you!

**Comment by gabek** [comment]
Maybe it's specifically a Twitter thing and just needs a twitter image assigned?

![image](https://user-images.githubusercontent.com/414923/209481915-cee5089f-5972-492b-a2eb-8af33f1e1e3b.png)

**Comment by gabek** [comment]
Or maybe it's pointing to a SVG and other sites can't render SVGs?

**Comment by germainelee** [comment]
It's a twitter thing. And Twitter likes name, not property. [Twitter Docs](https://developer.x.com/en/docs/twitter-for-websites/cards/overview/markup)

Edit: Found a handy site [OpenGraph](https://www.opengraph.xyz/url/https%3A%2F%2Fowncast.online) to copy paste the meta tags

**Comment by gabek** [comment]
What do you mean by liking name and not property? Could you give an example of what needs to be changed?

**Comment by germainelee** [comment]
According to Twitter docs, the **property** `<meta property="twitter:title" content="{{.Name}}">` should be **name**
```twitter
<meta name="twitter:title" content="{{.Name}}">
```

this too: `<meta property="twitter:description" content="{{.Summary}}">`
```twitter
<meta name="twitter:description" content="{{.Summary}}">
```

Here are some more example of Twitter meta tags: `<meta property="twitter:image" content="{{.Image}}">`
```twitter
<meta name="twitter:image" content="{{.Image}}">`
```

Twitter wanna be different selectively: `<meta property="twitter:card" content="player" />` 
```twitter
<meta name="twitter:card" content="player">
```

And these should be changed... to name
```code block 
<meta property="twitter:player" content='{{.RequestedURL}}embed/video' />
<meta property="twitter:player:width" content="560" />
<meta property="twitter:player:height" content="315" />
```

**Comment by gabek** [comment]
We could leave `property` for backwards compatibility and also add the `name` property.

**Comment by mahmed2000** [comment]
The twitter meta tags are already set with `name` on the actual page for owncast.online (I think from here: https://github.com/owncast/owncast.github.io/blob/a8e8fabb00607f5724b23163d62af918f7078b41/layouts/partials/head/twitter_cards.html#L3?). The problem seems to instead be the relative urls which are invalid. Most sites have some checking to fix invalid links but the "correct" way is with absolute urls. `og:url` is also incorrect so maybe that's why twitter is throwing a fit?

Easiest way to check would be to check the image source in the twitter post HTML.

https://github.com/jitsi/jitsi-meet/issues/6031

<blockquote><img src="https://opengraph.githubassets.com/42295f385ba85f774cfaa50d6f5cced506fdd3a2a696aaabd21825caf05a23b9/owncast/owncast.github.io" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/owncast/owncast.github.io/blob/a8e8fabb00607f5724b23163d62af918f7078b41/layouts/partials/head/twitter_cards.html">owncast.github.io/layouts/partials/head/twitter_cards.html at a8e8fabb00607f5724b23163d62af918f7078b41 · owncast/owncast.github.io</a></strong></div><div>Owncast's public facing web site.  Documentation and info. - owncast/owncast.github.io</div></blockquote>

**Comment by doug-wade** [comment]
I was looking for an issue for my first contribution, so I went poking around at the new docsaurus site to see if this was still worth addressing, and I think [we're gonna get this one for free with the migration](https://docusaurus.io/docs/seo). Here's what I see in the console when running locally:

<img width="823" height="56" alt="screenshot of opengraph meta tags showing the use of the fully-qualified owncast dot online domain" src="https://github.com/user-attachments/assets/c9bc41ad-adc9-4469-97f6-1499c8a8959d" />

**Comment by gabek** [comment]
Yeah, I think that's absolutely true. Shouldn't be something that we need to worry about going forward.
